### PR TITLE
Adds knight armour to the game that can be made from any material

### PR DESCRIPTION
**Pull request by ghost**
:cl: qustinnus
 add: adds Knight's Armour made out of any materials
 add; new ruin found in lavaland protected by dark wizards, I wonder what they're guarding
/:cl:
hello im taking a small break from my hiatus by making this pr. It's basically /datum/material knight armor which you can get by finding a design disk which I made a small ruin for.

yes this means diamond armor.

image

### CODE REVIEW

**Comment by zxaber** [comment]
What does plasma armor do? Does it release plasma gas when struck?

**Comment by Qustinnus** [comment]
plasma is probably the shittiest because it explodes on contact with fire.

**Comment by Qustinnus** [comment]
i might make it also release gas when punched, though.

File: code/datums/materials/_material.dm [hunk 1]
```diff
@@ -24,6 +24,8 @@ Simple datum which is instanced once per type and is used for every object of sa
 	var/strength_modifier = 1
 	///This is a modifier for integrity, and resembles the strength of the material
 	var/integrity_modifier = 1
+	///Armor modifiers, multiplies an items normal armor vars by these amounts.
+	var/armor_modifiers = list("melee" = 1, "bullet" = 1, "laser" = 1, "energy" = 1, "bomb" = 1, "bio" = 1, "rad" = 1, "fire" = 1, "acid" = 1)
```
Comment (optimumtact) [inline code/datums/materials/_material.dm]: gonna be impossible to balance

Comment (Qustinnus) [reply code/datums/materials/_material.dm]: I don't think you should be able to build any armor out of any material, this is more for edge-cases such as this one, or things that we can balance around this due to them only using this system, and thus having easily adjustable base numbers (walls, windows, doors)

Comment (PKPenguin321) [reply code/datums/materials/_material.dm]: Yeah as far as I can tell this isn't applicable to any armors except the one being added in this PR, balance shouldn't be a concern

**Comment by Shapsy** [comment]
are you planning on adjusting these numbers, cause right now this looks like printable riot gear++ that you can print at any lathe and reaches 75% full body melee resist with titanium (85% with sec jumpsuit), or the 90% armor cap if you don't care about radiation with uranium.

**Comment by Qustinnus** [comment]
ya they're not rly permanent i just wanted to see if it worked

**Comment by PKPenguin321** [comment]
having variable values on different materials like this gives me a good Rimworld vibe that I really dig, so I'm looking forward to seeing this expanded on

**Comment by Qustinnus** [comment]
@PKPenguin321 just wait until you see plasma armor and hold a lighter next to it ingame

**Comment by Qustinnus** [comment]
https://streamable.com/vl5e8 meme vid. I nerfed the amount of range plasma mat explosions have per unit 2.5 times after seeing this kek. the explosions are smaller now so dont worry

File: code/modules/research/designs/misc_designs.dm [hunk 2]
```diff
@@ -409,6 +409,37 @@
...
+	id = "knight_armour"
+	build_type = AUTOLATHE
+	materials = list(MAT_CATEGORY_RIGID = 10000)
+	build_path = /obj/item/clothing/suit/armor/riot/knight/greyscale
+	category = list("Equipment")
+	departmental_flags = DEPARTMENTAL_FLAG_ALL //Anyone can be a knight if they try hard enough
+
+/datum/design/knight_helmet
+	name = "Knight Helmet"
+	desc = "A royal knight's favorite hat. If you hold it upside down it's actually a bucket.."
```
Comment (MrPerson) [inline code/modules/research/designs/misc_designs.dm]: I know it's nitpicky but there's two periods here.

**Comment by optimumtact** [comment]
this is definitely going to be a mess of balance

**Comment by Qustinnus** [comment]
what are you worried about in particular? I don't think material datums will ever be all encompassing of all armors because those wont use those. Using material datums for items will only really happen for specific items where we control that to happen. Specifically for more joking items like this. I feel like if balance for this armor piece in particular is a concern, we can nerf it so that even the best version isn't stronger than sec stuff.

For example, sec armor etc wont ever become available as something that you can use datum mats on, so the modifiers should never affect the meta beyond when we specifically want it to.

**Comment by GuyonBroadway** [comment]
I say stop caring about moba levels of balancing down to 0.05 of a second and embrace the chaos and fun.

**Comment by Qustinnus** [comment]
image

**Comment by CRITAWAKETS** [comment]
maybe you should actually sprite instead of doing lazy compression of the original icon because right now runite looks awful

also tfw what the wizards are guarding is runite and not silverlight smh

File: code/datums/materials/basemats.dm [hunk 3]
```diff
@@ -75,16 +78,17 @@
 	name = "plasma"
 	id = "plasma"
 	desc = "Isn't plasma a state of matter? Oh whatever."
-	color = "#c716b8"
+	color = "#eb80f2"
 	categories = list(MAT_CATEGORY_ORE = TRUE, MAT_CATEGORY_RIGID = TRUE)
 	sheet_type = /obj/item/stack/sheet/mineral/plasma
 	coin_type = /obj/item/coin/plasma
+	armor_modifiers = list("melee" = 1.4, "bullet" = 0.7, "laser" = 0, "energy" = 1.2, "bomb" = 0, "bio" = 1.2, "rad" = 1, "fire" = 0, "acid" = 0.5)
```
Comment (ATH1909) [inline code/datums/materials/basemats.dm]: it'd be funny if this had a negative fire modifier

File: code/datums/materials/basemats.dm [hunk 4]
```diff
@@ -60,10 +62,11 @@
 	categories = list(MAT_CATEGORY_ORE = TRUE, MAT_CATEGORY_RIGID = TRUE)
 	sheet_type = /obj/item/stack/sheet/mineral/uranium
 	coin_type = /obj/item/coin/uranium
+	armor_modifiers = list("melee" = 2, "bullet" = 2, "laser" = 0.5, "energy" = 0.5, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 1, "acid" = 1)
```
Comment (ATH1909) [inline code/datums/materials/basemats.dm]: these melee and bullet armor values seem a bit high, even with the radioactive downside to armors made out of this

File: code/game/objects/items/stacks/sheets/mineral.dm [hunk 5]
```diff
@@ -371,6 +371,20 @@ GLOBAL_LIST_INIT(adamantine_recipes, list(
...
+ */
+
+/obj/item/stack/sheet/mineral/runite
+	name = "Runite"
+	desc = "Rare material found in distant lands."
+	singular_name = "runite bar"
+	icon_state = "sheet-runite"
+	item_state = "sheet-runite"
+	materials = list(/datum/material/runite=MINERAL_MATERIAL_AMOUNT)
+	merge_type = /obj/item/stack/sheet/mineral/runite
```
Comment (ATH1909) [inline code/game/objects/items/stacks/sheets/mineral.dm]: runite golems when?

File: code/modules/mob/living/simple_animal/hostile/dark_wizard.dm [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+	aggro_vision_range = 9
+	turns_per_move = 5
+	gold_core_spawnable = HOSTILE_SPAWN
+	faction = list(ROLE_WIZARD)
+	do_footstep = TRUE
+	weather_immunities = list("lava","ash")
+	minbodytemp = 0
+	maxbodytemp = INFINITY
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)	
+	loot = list(/obj/effect/decal/remains/human)
```
Comment (ATH1909) [inline code/modules/mob/living/simple_animal/hostile/dark_wizard.dm]: aw, c'mon, give 'em real spells

we have simplemob wizard (and construct) NPCs that cast spells (including fireball, blink, etc.); at least add a boss wizard or something who can cast "real" spells (sort of like a much weaker version of the paper wizard boss?)

Comment (Qustinnus) [reply code/modules/mob/living/simple_animal/hostile/dark_wizard.dm]: I dont really want them to be powerful and I dislike the wizard simple mobs in other ruins

File: code/datums/materials/basemats.dm [hunk 7]
```diff
@@ -130,6 +140,16 @@
 	strength_modifier = 1.3
 	categories = list(MAT_CATEGORY_ORE = TRUE, MAT_CATEGORY_RIGID = TRUE)
 	sheet_type = /obj/item/stack/sheet/mineral/titanium
+	armor_modifiers = list("melee" = 1.35, "bullet" = 1.3, "laser" = 1.3, "energy" = 1.25, "bomb" = 1.25, "bio" = 1, "rad" = 1, "fire" = 0.7, "acid" = 1)
+
+/datum/material/runite
```
Comment (Fikou) [inline code/datums/materials/basemats.dm]: shouldnt sheet_type be in this material, with the new sheets you added

**Comment by Ratvar** [comment]
Why Is This A Good Thing For The Game?
What's the point of resource balancing if you can pick the best one easily?

File: code/datums/materials/basemats.dm (hunk 7 above)
Comment (Qustinnus) [reply code/datums/materials/basemats.dm]: My bad, thanks.

File: code/datums/materials/basemats.dm (hunk 4 above)
Comment (Qustinnus) [reply code/datums/materials/basemats.dm]: You'll die of radiation before you can do much.

File: code/datums/materials/basemats.dm (hunk 3 above)
Comment (Qustinnus) [reply code/datums/materials/basemats.dm]: You'd blow up before it matters

**Comment by Qustinnus** [comment]
@critawakets wdym compression? It's a grey scale sprite it's supposed to be automatic that's kind of the point lol

**Comment by Qustinnus** [comment]
@ratvar materials will have more upsides and downsides once I get to them

File: code/datums/materials/basemats.dm (hunk 4 above)
Comment (Shapsy) [reply code/datums/materials/basemats.dm]: Unless you're a radiation proof species?

Comment (nemvar) [reply code/datums/materials/basemats.dm]: Epic plasmeme gamers

File: code/datums/components/wearercrosser.dm [hunk 8]
```diff
@@ -0,0 +1,14 @@
+///Calls crossed on the worn item when the mob wearing it gets crossed
+/datum/component/wearertargeting/wearer_crosser
+	signals = list(COMSIG_MOVABLE_CROSSED)
+	mobtype = /mob/living/carbon
+	proctype = .proc/propogate_crossed
+	var/obj/item/item
```
Comment (nemvar) [inline code/datums/components/wearercrosser.dm]: You will have to null this on Destroy

File: code/datums/materials/basemats.dm (hunk 4 above)
Comment (Qustinnus) [reply code/datums/materials/basemats.dm]: oh yeah fuck lmao lemme smash that nerf button

File: code/datums/materials/basemats.dm [hunk 9]
```diff
@@ -109,17 +113,23 @@
...
 	coin_type = /obj/item/coin/bananium
+	armor_modifiers = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 100, "bio" = 0, "rad" = 0, "fire" = 10, "acid" = 0) //Clowns cant be blown away.
 
 /datum/material/bananium/on_applied(atom/source, amount, material_flags)
 	. = ..()
 	source.AddComponent(/datum/component/squeak, list('sound/items/bikehorn.ogg'=1), 50)
 	source.AddComponent(/datum/component/slippery, min(amount / 10, 80))
 
+/datum/material/bananium/on_applied_obj(var/obj/o, amount, material_flags)
+	o.AddComponent(/datum/component/wearertargeting/wearer_crosser, list(SLOT_HEAD, SLOT_SHOES, SLOT_BACK, SLOT_W_UNIFORM, SLOT_WEAR_SUIT))
```
Comment (nemvar) [inline code/datums/materials/basemats.dm]: This is scuffed. wearertargeting only allows items but this tries to add it to any obj.

**Comment by nemvar** [comment]
Alright, three more complaints.
Map wasn't run through map merger.
Wiznerd has weird sprites. Only looks to be recoloured in one direction.
udzdva dw
Are you sure we can use the sprites for the runite bars?

**Comment by Qustinnus** [comment]
yeah i broke the mapmerger so i need to try and fix it later but cba yesterday.

also fuck i forgot that wizards had dirs gonna kms lmao

**Comment by Mickyan** [comment]
You could probably bypass most concerns about balance if the average joe couldn't just print these at any lathe

We have framework for learning crafting recipes that hasn't been used yet(#43296), perhaps this could be a thing for ash lizards and/or maybe one of the options for the curator's beacon could be a skillbook to learn this

**Comment by Qustinnus** [comment]
you can only get it if you find the disk in lavaland.

**Comment by Mickyan** [comment]
I'm a big dummy and can't read

A recipe book would still be more thematically appropriate than a design disk though

**Comment by Qustinnus** [comment]
yeah but autolathes are the only machines that support material crafting atm

File: code/datums/materials/basemats.dm (hunk 3 above)
Comment (ATH1909) [reply code/datums/materials/basemats.dm]: Still, even the plasma medal gives a little bit of fire vulnerability; I don't see why you can't just give this a -0.1 fire armor modifier or something

**Comment by NoxVS** [comment]
> yeah but autolathes are the only machines that support material crafting atm

Barely. Im pretty sure right now the only way to remove non metal or glass materials from an autolathe is to deconstruct it. Why not just make a new machine to handle things of this type rather than try to shove it all in to a machine that doesn't support it.

**Comment by Qustinnus** [comment]
OK you say barely but it's still the only machine to support it. If you want it to be its own machine go code it.

**Comment by Shadowflame909** [comment]
>Slip Armor

epic

>Armor that explodes you

noob bait

**Comment by FantasticFwoosh** [comment]
I don't really care for balancing the suit as its a fun little thing people can add to as we go along, but using and fleshing out [mythril](https://github.com/tgstation/tgstation/blob/a4d953f75a6bcf0fbbca08c3b48cdc1c2198b605/code/game/objects/items/stacks/sheets/mineral.dm#L377) which we already have would probably be a better use than making up a material on the spot.

Unless runite is a magic-tech ORM composite of adamantine and mythril that is, maybe we can do more with this later.

**Comment by MCterra10** [comment]
Cultist armors made from the respective cult metals would be lit, but they should only be wearable by cultists

**Comment by bigfatbananacyclops** [comment]
where is the diamond sword

**Comment by actioninja** [comment]
conflict

**Comment by Qustinnus** [comment]
conflicts fixed

**Comment by optimumtact** [comment]
You know what you guys are really fucking me off, stop saying balance doesn't matter, it does fucking matter, and when I have to fucking merge nerfs later, it's not your ass that is on the line, it's some other poor coder who's willing to do the work.

**Comment by Qustinnus** [comment]
i don't mind nerfing things at all. if someone tells me something should be nerfed (and it isn't a wesoda-esque attack) I'm not against nerfing it, I've nerfed mood, I was trying to nerf hygiene, and I already nerfed this PR, this armour is not particularly great, takes some effort, and takes materials. Most you'll get from it is nice looking armor.

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/living.dm [hunk 10]
```diff
@@ -534,6 +534,14 @@
 	if(should_update_mobility)
 		update_mobility()
 
+/mob/living/Crossed(atom/movable/AM)
+	. = ..()
+	for(var/i in get_equipped_items())
+		var/obj/item/item = i
+		item.Crossed(AM)
```
Comment (ninjanomnom) [inline code/modules/mob/living/living.dm]: As we talked about you should make a new proc for this hook, WearerCrossed() or something

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/living.dm (hunk 10 above)
Comment (Qustinnus) [reply code/modules/mob/living/living.dm]: Oh I didn't see that part. One second

**Review by ninjanomnom — APPROVED** [review]

**Comment by Qustinnus** [comment]
YEESSSSS

**Pull request merged**